### ISSUE TITLE
Train multi-output Random Forest for joint environmental triad prediction (broad+local+medium)

### ISSUE THREAD

**Issue by turbomam** [contextualizer-ai/env-embeddings#52]
## Current Approach (Issue #49)

Training 3 **independent** Random Forest classifiers:
1. `rf_broad` predicts `env_broad_scale` from GE embeddings
2. `rf_local` predicts `env_local_scale` from GE embeddings  
3. `rf_medium` predicts `env_medium` from GE embeddings

**Problem**: Ignores dependencies between scales.

## Motivation

The environmental triad has **hierarchical structure**:
- `env_broad_scale` (biome) constrains `env_local_scale` (habitat)
- Example: If broad="marine biome", local can't be "desert"

**Research questions**:
1. Does predicting all 3 scales **jointly** improve accuracy?
2. Can we exploit hierarchical constraints?
3. Which triads are most/least predictable from satellite imagery?

### Approach 1: Multi-Output Random Forest

Use sklearn's native multi-output support:

```python
from sklearn.multioutput import MultiOutputClassifier

# Create target matrix (n_samples, 3)
y_multi = df[['env_broad_scale', 'env_local_scale', 'env_medium']].values

# Train single model predicting all 3 scales
rf_multi = MultiOutputClassifier(
    RandomForestClassifier(n_estimators=100, max_depth=10)
)
rf_multi.fit(X_train, y_multi_train)

# Predict all 3 at once
y_pred_multi = rf_multi.predict(X_test)  # shape: (n_test, 3)
```

**Pros**: Simple, sklearn native, shares feature importance across scales
**Cons**: May not capture scale dependencies well

### Approach 2: Chained Classifiers

Use predicted broad_scale as feature for predicting local/medium:

```python
from sklearn.multioutput import ClassifierChain

# Chain: broad → local → medium
chain = ClassifierChain(
    RandomForestClassifier(n_estimators=100),
    order=[0, 1, 2]  # broad, local, medium
)
chain.fit(X_train, y_multi_train)
```

**Pros**: Captures dependencies, hierarchical
**Cons**: Error propagation (wrong broad → wrong local)

### Approach 3: Joint Encoding

Treat entire triad as single categorical variable:

```python
# Create joint labels
df['triad'] = df['env_broad_scale'] + '|' + df['env_local_scale'] + '|' + df['env_medium']

# Train single classifier
# Unique triads: ~1000-5000 classes (vs 500-600 per scale)
rf_joint = RandomForestClassifier(...)
rf_joint.fit(X_train, y_triad_train)
```

**Pros**: Learns triad patterns directly, no independence assumption
**Cons**: Huge class space, data sparsity, doesn't generalize to unseen triads

### Per-Scale Accuracy
- Compare to independent classifiers from #49
- Does joint training help or hurt individual scales?

### Triad-Level Accuracy
```python
# All 3 scales must match
triad_correct = (y_multi_test == y_pred_multi).all(axis=1)
triad_accuracy = triad_correct.mean()
```

### Partial Triad Accuracy
- 2 of 3 correct
- 1 of 3 correct  
- 0 of 3 correct

### Constraint Violations
For Approach 2/3, check:
- How often does predicted triad violate ENVO hierarchy?
- Example: broad="marine" but local="desert" (impossible)

## Implementation Tasks

- [ ] Create `notebooks/random_forest_joint_triad_prediction.ipynb`
- [ ] Implement Approach 1 (Multi-Output RF)
- [ ] Implement Approach 2 (Chained Classifiers)
- [ ] Implement Approach 3 (Joint Encoding) - optional
- [ ] Evaluation:
  - [ ] Compare to independent classifiers (baseline from #49)
  - [ ] Per-scale accuracy comparison
  - [ ] Triad-level accuracy
  - [ ] Execution time comparison
  - [ ] Feature importance: do scales share important features?
- [ ] Visualizations:
  - [ ] Triad confusion matrix (heatmap)
  - [ ] Most/least predictable triads
  - [ ] Constraint violation analysis
- [ ] Document findings in notebook markdown

## Expected Results

**Hypothesis**: Joint prediction improves triad-level accuracy but may slightly hurt individual scales.

**Baseline (independent, from #49)**:
- broad_accuracy: ~55%
- local_accuracy: ~53%
- medium_accuracy: ~50%
- triad_accuracy (all 3 correct): ~25% (0.55 × 0.53 × 0.50 if independent)

**Expected (multi-output)**:
- Per-scale: ~50-55% (similar or slightly lower)
- Triad: ~30-35% (5-10% improvement from constraints)

**Best case**: Chained classifier with hierarchical constraints improves triad accuracy to 40%+

## Related

- #49 - Independent classifiers (baseline for comparison)
- #51 - Ontology-aware evaluation (complementary)
- #31 - Original classifier idea

## Priority

**Medium** - Interesting research question, moderate implementation effort, may or may not improve accuracy (experiment needed).

## Notes

- Run **after** #49 completes to have baseline
- Start with Approach 1 (simplest)
- Approach 2 most promising for scientific insight
- Approach 3 may fail due to data sparsity (try only if others succeed)

**Comment by turbomam** [comment]
joint embeddings is a well characterized problem

but this still a low priority
